File: data/imported/knowledge/cells/cpp/merged/snippets/snippet_075.cpp
TEST(WorkbookPortingTests, LoadXlsxFilePreservesCellsAndFormula)
{
    TempDir temp;
    const auto path = temp.Path("roundtrip.xlsx");

    Workbook workbook;
    PopulateBasicWorkbook(workbook);
    workbook.Save(path.string());

    Workbook loaded(path.string());
    auto& cells = loaded.GetWorksheets()["Data"].GetCells();

    EXPECT_EQ("alpha", cells["A1"].GetStringValue());
    EXPECT_EQ(42, cells["B1"].GetValue().AsInteger());
    EXPECT_TRUE(cells["C1"].GetValue().AsBool());
    EXPECT_EQ(DateTime(2024, 5, 6, 7, 8, 9), cells["D1"].GetValue().AsDateTime());
    EXPECT_EQ("=B1*2", cells["E1"].GetFormula());
    EXPECT_EQ(21, cells["E1"].GetValue().AsInteger());
}

File: data/imported/knowledge/cells/java/merged/snippets/snippet_075.java
@Test
    void CV_25_largeDoubleScientificNotation() {
        Workbook wb = new Workbook();
        Cell cell = wb.getWorksheets().get(0).getCells().get("A1");
        double val = 6.02214076E+23;
        cell.putValue(val);
        assertEquals(CellValueType.NUMBER, cell.getType());
        assertEquals(val, (Double) cell.getValue(), 1e9);
    }